vicare/climate.py

```diff
@@ -136,47 +136,63 @@ def __init__(self, name, api, heating_type):
+                _supply_temperature = self._api.getSupplyTemperature()
+            except PyViCareNotSupportedFeatureError as e:
+                _LOGGER.error("Feature not supported " + str(e))
+
+            if _room_temperature is not None:
                 self._current_temperature = _room_temperature
-            elif _supply_temperature != PYVICARE_ERROR:
+            elif _supply_temperature is not None:
                 self._current_temperature = _supply_temperature
             else:
                 self._current_temperature = None
-            self._current_program = self._api.getActiveProgram()
 
-            # The getCurrentDesiredTemperature call can yield 'error' (str) when the system is in standby
-            desired_temperature = self._api.getCurrentDesiredTemperature()
-            if desired_temperature == PYVICARE_ERROR:
-                desired_temperature = None
+            try:
+                self._current_program = self._api.getActiveProgram()
+            except PyViCareNotSupportedFeatureError as e:
+                _LOGGER.error("Feature not supported " + str(e))
+
+            try:
+                desired_temperature = self._api.getCurrentDesiredTemperature()
+            except PyViCareNotSupportedFeatureError as e:
+                _LOGGER.error("Feature not supported " + str(e))
 
             self._target_temperature = desired_temperature
 
-            self._current_mode = self._api.getActiveMode()
+            try:
+                self._current_mode = self._api.getActiveMode()
+            except PyViCareNotSupportedFeatureError as e:
+                _LOGGER.error("Feature not supported " + str(e))
 
             # Update the generic device attributes
             self._attributes = {}
-            self._attributes["room_temperature"] = _room_temperature
+            # TODO: catch exception for each attribute:
+            #  self._attributes["room_temperature"] = _room_temperature
             self._attributes["active_vicare_program"] = self._current_program
             self._attributes["active_vicare_mode"] = self._current_mode
             self._attributes["heating_curve_slope"] = self._api.getHeatingCurveSlope()
             self._attributes["heating_curve_shift"] = self._api.getHeatingCurveShift()
-            self._attributes[
-                "month_since_last_service"
-            ] = self._api.getMonthSinceLastService()
-            self._attributes["date_last_service"] = self._api.getLastServiceDate()
-            self._attributes["error_history"] = self._api.getErrorHistory()
-            self._attributes["active_error"] = self._api.getActiveError()
 
             # Update the specific device attributes
-            if self._heating_type == HeatingType.gas:
-                self._current_action = self._api.getBurnerActive()
-
-            elif self._heating_type == HeatingType.heatpump:
-                self._current_action = self._api.getCompressorActive()
+            try:
```

Comment:
It would be nice to have a "getFeatureSafe" method where you can pass the assignment and it wraps it with try-except. But I don't know how to do this in Python ;)

Reply:
I tried to make it nicer with contextmanager but not sure it's really much better.
Anyway I think we should get rid of the error logs since it's heavily dependant on the heating device what parameters are available and we might end up spamming logs.
Maybe in the future we should check if a certain datapoint is available and provide the sensor only then.